Make collect-info multi-arch compatible, providing arm64 support

michaelsembwever opened the pull request.

rzvoncek commented on Makefile:
```diff
@@ -1,10 +1,13 @@
 	@rm -f collector/collect-info
-	@cd collector ; docker run --rm -v $$PWD:/volume -w /volume -t clux/muslrust rustc --target x86_64-unknown-linux-musl rust-commands/*.rs ; cd -
-	@test -f collector/collect-info
+	@cd collector ; docker run --rm --platform linux/arm64 -v /usr/bin/qemu-aarch64-static:/usr/bin/qemu-aarch64-static -v $$PWD:/volume -w /volume -t clux/muslrust rustc --target aarch64-unknown-linux-musl rust-commands/*.rs ; mv collect-info collect-info.aarch64-unknown-linux-musl ; cd -
```
the `cd -` at the end just undoes the `cd collector` from the beginning?

michaelsembwever replied:
intentional.

michaelsembwever commented:
> The tests seem to have a bad time tho.

happens sometimes.  restarted them.

michaelsembwever commented:
Clean CI.

Additional adjustments to integration tests:
 - Had to remove the macos step, as the macos runners are now arm64 mX based and don't support docker.
 - Upgraded cass-operator, Cassandra, and DSE versions used in the tests.  
 - Reduced DSE tests to single node.
 - Had to use `--force-conflicts --server-side` on cass-operator, see https://github.com/k8ssandra/k8ssandra/issues/1380
 - Added iproute2 to test images (to get `ip` command)
 - A darwin arm64 `collect-info` is built if the local machine is such and `ructc` is on the path, and this is used when running `ds-collector`, on the assumption this is a mac mX machine running integration tests.  A clear message is printed if this assumption is wrong and how to remedy it.
 - The use of `df` command is made macos friendly in `ds-collector` (for the above point).

Also added the licence header to ds-collector script.
